---
title: "Bess Tamoxifen Analysis"
output: html_document
date: "`r Sys.Date()`"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo=FALSE, warning=FALSE)

library(tidyverse)
library(vegan)
library(ggpubr)
library(RColorBrewer)
library(knitr)
library(kableExtra)
library(ggrepel)
library(readxl)
library(Hmisc)

options(scipen=999)

setwd("C:/Users/juli.avelar-barragan/Desktop/Project_bioinformatics/Bess_lab_tamoxifen/")

OTU_table = read.delim("combined_bracken_species.txt", check.names = F, row.names = 1) %>%
  select(contains("num")) %>%
  rename_with(~gsub("_species.bracken_num", "", .x)) %>%
  filter(rowSums(.) != 0)
OTU_table = OTU_table[!rownames(OTU_table) == "Homo sapiens",]

Readcounts = read.csv("Readinfo.csv", check.names = F)
```

# Project information:  
A pharmacokinetic study performed by Dr. Elizabeth Bess in mice highlighted the gut microbiome's role in tamoxifen bioactivation. Enzyme lysates from the fecal samples of nine individuals were analyzed for their ability to hydrolyze glucuronidated tamoxifen metabolites, revealing high variability in hydrolyzing nitrogen-linked glucuronides and less variability with oxygen-linked glucuronides. The nine fecal samples were sequenced using whole genome shotgun sequencing (WGS) to characterize microbial Î²-glucuronidase (GUS) genes, as their substrate specificity may explain the inter-individual differences in enzyme activity.  

# Read count information per sample:  
Raw - # of reads from the sequencer.  
QC - # of reads after the removal of PhiX, adapter, and low quality reads.  
Host removed - # of reads after removing reads which map to the human genome.  
```{r echo=FALSE}
rc_df = 
  Readcounts %>%
  mutate(`Reads removed by QC` = Raw - QC,
         `% removed by QC` = round((Raw - QC)/Raw*100, digits = 1),
         `Reads removed from host` = QC - Host_removed,
         `% removed from host` = round((QC - Host_removed)/QC*100, digits = 3)) 

rc_df %>%
  kbl() %>%
  kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive"))
```

Table of averages.  
```{r echo=FALSE}
rc_df %>%
  pivot_longer(!SampleID) %>%
  group_by(name) %>%
  summarise(Mean = round(mean(value)),
            `Std. dev.` = round(sd (value))) %>%
  arrange(desc(Mean)) %>%
  kbl() %>%
  kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive"))
```

# Percentage of reads classified with Kraken2:  
```{r echo=FALSE}
#Data pulled from Kraken2 output logs on the HPC.
unknown_df = data.frame(
  "SampleID" = c("14", "17", "19", "20", "2126", "22a", "22b", "25", "2621", "27", "Stan"),
  "Classified" = c(24671608, 21667846, 21449550, 34139213, 8772618, 8491075, 20487188, 14859100, 14811562, 12939547, 10149406),
  "Unclassified" = c(32210414, 62875189, 23696489, 43507132, 12323363, 12527792, 31296255, 26457997, 29812345, 23602613, 14855679))

unknown_df = unknown_df %>%
  mutate(
    Total = Classified + Unclassified,
    `% classified` = round(Classified/Total*100, digits = 1),
    `% unclassified` = round(Unclassified/Total*100, digits = 1)
  )
  
unknown_df %>%
  kbl() %>%
  kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive"))
```

```{r echo=FALSE, warning=FALSE}
unknown_plot_df = unknown_df %>%
  pivot_longer(!SampleID) %>%
  filter(name %in% c("% unclassified", "% classified"))

ggplot(data = unknown_plot_df) +
  aes(x = name, y = value, label = SampleID) +
  geom_boxplot(outlier.shape = "none", color = "lightblue", size = 1) +
  geom_text_repel() +
  theme_bw() +
  labs(y = "Percentage", x = NULL)
```

# Alpha diversity:  
Performed at the species level.  
```{r echo=FALSE}
alpha_df = as.data.frame(diversity(x = OTU_table, MARGIN = 2)) %>%
  rename(Shannon = `diversity(x = OTU_table, MARGIN = 2)`) %>%
  rownames_to_column(var = "SampleID") %>%
  mutate(Richness = specnumber(OTU_table, MARGIN = 2)) 

alpha_df %>%
  kbl() %>%
  kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive"))
```

# Beta diversity
Non-metric multidemensional scaling (NMDS) plot using species taxonomy and the Bray-Curtis metric.  
```{r include=FALSE}
nmds = metaMDS(t(OTU_table), trymax = 1024, parallel = 8, k = 2)
nmds_points = as.data.frame(nmds$points) %>%
  rownames_to_column(var = "SampleID")
```

```{r echo=FALSE, warning=FALSE}
ggplot(data = nmds_points) +
  aes(x = MDS1, y = MDS2) +
  geom_text(aes(label = SampleID), color = "black", size = 5, alpha = 0.7) +
  theme_bw() +
  annotate("text", x = Inf, y = Inf, label = bquote("Stress ="~.(round(nmds$stress, digits = 2))), hjust = 1, vjust = 1)
```

Stacked bar plot showing the taxonomic composition.  
```{r echo=FALSE, message=FALSE, warning=FALSE}
# Define a vector of taxonomy levels
taxonomy_levels <- c("phylum", "class", "order", "family", "genus", "species")
plots_list <- list()

# Loop through each taxonomy level
for (level in taxonomy_levels) {
  
  # Construct the file name based on the current taxonomy level
  file_name <- paste0("combined_bracken_", level, ".txt")
  
  # Read data from the current file
  bp_OTU_table <- read.delim(file_name, check.names = FALSE, row.names = 1) %>%
    select(contains("num")) %>%
    rename_with(~gsub(paste0("_", level, ".bracken_num"), "", .x)) %>%
    filter(rowSums(.) != 0)
  
  # Continue with your existing data processing steps
  bp_OTU_table <- decostand(bp_OTU_table, MARGIN = 2, method = "total") %>%
    rownames_to_column(var = "Taxonomy")
  
  top_10 <- bp_OTU_table %>%
    rowwise(Taxonomy) %>% 
    summarise(mean = mean(c_across(everything()))) %>% 
    ungroup %>% 
    slice_max(mean, n = 10)
  
  barplot_df <- bp_OTU_table %>% 
    pivot_longer(!Taxonomy) %>%
    mutate(Taxonomy = if_else(Taxonomy %in% top_10$Taxonomy, Taxonomy, "Other")) %>%
    group_by(Taxonomy, name) %>%
    summarise(value = sum(value))
  
  barplot_df <- rbind(barplot_df[!(barplot_df$Taxonomy == "Other"),],barplot_df[(barplot_df$Taxonomy == "Other"),]) #Move other to bottom
  barplot_df$Taxonomy <- factor(barplot_df$Taxonomy, levels = unique(barplot_df$Taxonomy)) #Fix the order
  
  Julio_color <- c("#003f5c", "#665191", "#d45087", "#ff7c43","#ffa600", "#7F0A57", "#CD9ABB", "#39A9AB", "#71CFC5", "#007947" ,"gray")
  
  bp_plot = ggplot(data = barplot_df) +
    aes(x = name, weight = value, fill = Taxonomy) +
    geom_bar() +
    theme_classic() +
    labs(x = NULL, y = "Relative abundance", fill = tools::toTitleCase(level)) +
    theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5)) +
    scale_fill_manual(values = Julio_color)
  
  plots_list[[paste0("plot_", level)]] <- bp_plot
}

plots_list
```

# GUS gene profiling

Methods:  
* Protein FASTA files from the supplement of https://doi.org/10.1016%2Fj.str.2017.05.003 were downloaded and DIAMOND was used to align sample reads to GUS genes at a minimum of 90% identity.  
* If multiple reads mapped to multiple GUS genes, the read count was divided by the total number of features. For example, if one read mapped to three GUS genes identically, the total read count for each GUS gene would increase by +1/3.  
* GUS gene counts were normalized to the number of reads per sample in reads per million (RPM).  
```{r echo=FALSE}
HMGC279 = read.delim("HMGC279_RPM_counts.txt", check.names = F, row.names = 1) %>%
  rename(`SRS020328.271517-mL2` = `SRS020328.271517-mL2Parabacteroides`,
         `SRS024132.332613-mL2` = `SRS024132.332613-mL2Bacteroides`)

HMGI3013 = read.delim("HMGI3013_RPM_counts.txt", check.names = F, row.names = 1)

GUS_summary = as.data.frame(rowSums(HMGC279)) %>%
  rename(`Normalized Read Counts (HMGC279)` = `rowSums(HMGC279)`) %>%
  mutate(`Richness (HMGC279)` = specnumber(HMGC279),
         `Normalized Read Counts (HMGI3013)` = rowSums(HMGI3013),
         `Richness (HMGI3013)` = specnumber(HMGI3013)) %>%
  rownames_to_column(var = "SampleID")

GUS_summary %>%
  kbl() %>%
  kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive"))
```

```{r warning=FALSE, include=FALSE}
HMGC279_Sequences <- read.delim("HMGC279_Sequences.txt", header=FALSE) %>%
  filter(str_detect(V1, ">")) %>%
  mutate(cleaned_column = str_extract(V1, "(?<=\\s).*"),
         sample_id = str_extract(V1, "^[^\\s]+") %>% str_remove("^>"))

HMGC279_species = HMGC279 %>%
  rownames_to_column("sample") %>%
  pivot_longer(cols = -sample, names_to = "feature", values_to = "abundance") %>%
  left_join(HMGC279_Sequences, by = join_by("feature" == "sample_id")) %>%
  group_by(sample, cleaned_column) %>%
  summarise(abundance = sum(abundance))

GUS_species_df = pivot_wider(data = HMGC279_species, id_cols = sample, names_from = cleaned_column, values_from = abundance) %>%
  column_to_rownames(var = "sample")

GUS_species_nmds = metaMDS(GUS_species_df)
species_nmds_vectors = envfit(GUS_species_nmds, GUS_species_df)

# Extract feature vectors and filter for significance
feature_coords = as.data.frame(scores(species_nmds_vectors, "vectors")) * ordiArrowMul(species_nmds_vectors)
feature_coords$pvals = species_nmds_vectors$vectors$pvals
feature_coords = filter(feature_coords, pvals < 0.05)
```

Ordination plot of GUS gene abundances.  
First is the summarized version of the HMGC279 data, which combines all abundances from the same species.  

```{r echo=FALSE}
ggplot() +
  geom_segment(data = feature_coords, 
               aes(x = 0, y = 0, xend = NMDS1, yend = NMDS2),
               arrow = arrow(length = unit(0.2, "cm")), color = "red", alpha = 0.2) +
  geom_text_repel(data = feature_coords, 
            aes(x = NMDS1, y = NMDS2, label = rownames(feature_coords)),
            hjust = -0.2, vjust = 1,  color = "gray30", size = 3) +
  geom_text(data = GUS_species_nmds$points, aes(x = MDS1, y = MDS2, label = rownames(GUS_species_nmds$points))) +
  theme_bw() +
  labs(title = "HMGC279 Summary Data", x = "MDS1", y = "MDS2") +
  annotate("text", x = Inf, y = Inf, label = bquote("Stress ="~.(round(GUS_species_nmds$stress, digits = 2))), hjust = 1, vjust = 1)
```

```{r warning=FALSE, include=FALSE}
GUS279_MDS = metaMDS(filter(HMGC279, !row.names(HMGC279) %in% c("2126", "22b", "2621", "Stan")))
HMGC279_vectors = envfit(GUS279_MDS, filter(HMGC279, !row.names(HMGC279) %in% c("2126", "22b", "2621", "Stan")))


# Extract feature vectors and filter for significance
feature_coords = as.data.frame(scores(HMGC279_vectors, "vectors")) * ordiArrowMul(HMGC279_vectors)
feature_coords$pvals = HMGC279_vectors$vectors$pvals
feature_coords = filter(feature_coords, pvals < 0.05) %>%
  rownames_to_column(var = "sample_id") %>%
  left_join(HMGC279_Sequences)
```

```{r warning=FALSE, include=FALSE}
GUS3013_sequences = read_excel("C:/Users/juli.avelar-barragan/Desktop/Project_bioinformatics/Bess_lab_tamoxifen/HMGI3013_Characteristics.xlsx", 
    sheet = "HMGI_gus_seqs_coALign_LP1LP2")

GUS3013_MDS = metaMDS(filter(HMGI3013, !row.names(HMGI3013) %in% c("2126", "22b", "2621", "Stan")))
HMGI3013_vectors = envfit(GUS3013_MDS, filter(HMGI3013, !row.names(HMGI3013) %in% c("2126", "22b", "2621", "Stan")))

# Extract feature vectors and filter for significance
feature_coords2 = as.data.frame(scores(HMGI3013_vectors, "vectors")) * ordiArrowMul(HMGI3013_vectors)
feature_coords2$pvals = HMGI3013_vectors$vectors$pvals
feature_coords2 = filter(feature_coords2, pvals < 0.05) %>%
  rownames_to_column(var = "id") %>%
  left_join(GUS3013_sequences) %>%
  filter(complete.cases(.))
```

Plots showing the top 10 most abundant GUS gene from the summarized HMGC279 data set.   
```{r echo=FALSE}
GUS_species_df2 = GUS_species_df %>%
  rownames_to_column(var = "Sample")

# Step 1: Create a function to identify the top 10 species per sample and create a long-format data frame
top_10_species_per_sample <- function(sample_data) {
  top_species <- sample_data %>%
    select(-Sample) %>%
    pivot_longer(cols = everything(), names_to = "Species", values_to = "Count") %>%
    arrange(desc(Count)) %>%
    slice(1:10)
  
  other_species <- sample_data %>%
    select(-Sample) %>%
    pivot_longer(cols = everything(), names_to = "Species", values_to = "Count") %>%
    filter(!Species %in% top_species$Species) %>%
    summarise(Species = "Other", Count = sum(Count))
  
  bind_rows(top_species, other_species) %>%
    mutate(Sample = sample_data$Sample)
}

# Step 2: Apply the function to each sample and combine the results
df_long <- GUS_species_df2 %>%
  group_by(Sample) %>%
  group_split() %>%
  map_dfr(top_10_species_per_sample)

# Step 3: Create the stacked barplots
plots <- df_long %>%
  mutate(Species = factor(Species, levels = c(setdiff(unique(Species), "Other"), "Other"))) %>%
  split(.$Sample) %>%
  map(~ ggplot(.x, aes(x = "", y = Count, fill = Species)) +
        geom_bar(stat = "identity", color = "black") +
        theme_bw() +
        theme(axis.ticks.x = element_blank()) +
        ylim(0, 550) +
        scale_fill_manual(values = Julio_color) +
        labs(x = NULL, y = "Count (RPM)", fill = "Taxonomy", title = unique(.x$Sample)))

# To display all plots (one per sample)
plots
```

# Correlations
No normalization was performed on the protein assay values (Based on standards). The three readings were averaged together. Samples 21 and 26 are missing due to not being able to assign the proper protein assay values to them.   
```{r include=FALSE}
Glucorunide_incubation_GUS = read.csv("Glucorunide_incubation_GUS.csv")

plot_names = Glucorunide_incubation_GUS %>% 
  select(SampleID, AltID) %>% 
  distinct(SampleID, AltID) %>%
  filter(complete.cases(.))

protein_assay_data = Glucorunide_incubation_GUS %>%
  filter(complete.cases(.)) %>%
  pivot_longer(!c("SampleID", "Condition")) %>%
  group_by(SampleID, Condition) %>%
  summarise(Value = mean(value)) %>%
  pivot_wider(id_cols = SampleID, names_from = Condition, values_from = Value)

create_heatmap <- function(GUS_df, protein_df) {
  # Prepare the data by merging and converting to the appropriate format
  tmp <- GUS_df %>%
    rownames_to_column(var = "SampleID") %>%
    inner_join(protein_df) %>%
    column_to_rownames("SampleID")
  
  # Calculate Spearman correlation
  tmp2 <- rcorr(as.matrix(tmp), type = "spearman")
  
  # Extract and transform the correlation coefficients
  tmp3 <- as.data.frame(tmp2$r) %>%
    select(`(E)-4-hydroxy tamoxifen O-b-D-glucuronide`, `N-Desmethyl-4-hydroxytamoxifen b-D-glucuronide (E/Z mixture)`) %>%
    filter(!row.names(.) %in% c("(E)-4-hydroxy tamoxifen O-b-D-glucuronide", "N-Desmethyl-4-hydroxytamoxifen b-D-glucuronide (E/Z mixture)")) %>%
    rownames_to_column(var = "Taxonomy") %>%
    pivot_longer(!Taxonomy, values_to = "Rho")
  
  # Extract and transform the p-values
  tmp4 <- as.data.frame(tmp2$P) %>%
    select(`(E)-4-hydroxy tamoxifen O-b-D-glucuronide`, `N-Desmethyl-4-hydroxytamoxifen b-D-glucuronide (E/Z mixture)`) %>%
    filter(!row.names(.) %in% c("(E)-4-hydroxy tamoxifen O-b-D-glucuronide", "N-Desmethyl-4-hydroxytamoxifen b-D-glucuronide (E/Z mixture)")) %>%
    rownames_to_column(var = "Taxonomy") %>%
    pivot_longer(!Taxonomy, values_to = "Pval") %>%
    left_join(tmp3, by = c("Taxonomy", "name")) %>%
    mutate(padj = p.adjust(.$Pval, method = "BH")) %>%
    filter(complete.cases(.))
  
  # Prepare the data for heatmap
  heatmap_df <- pivot_wider(data = tmp4, id_cols = Taxonomy, names_from = name, values_from = Rho) %>%
    column_to_rownames(var = "Taxonomy")
  
  # Perform hierarchical clustering
  clust_order1 <- hclust(dist(heatmap_df, method = "euclidean"))
  tmp4$Taxonomy <- factor(tmp4$Taxonomy, levels = rownames(heatmap_df)[clust_order1$order])
  
  # Generate the heatmap
  plot = ggplot(data = tmp4) +
    geom_tile(aes(x = name, y = Taxonomy, fill = Rho)) +
    scale_fill_gradient2(mid = "#FBFEF9", low = "#0C6291", high = "#A63446", limits = c(-1, 1)) +
    theme_classic() +
    labs(x = NULL, y = NULL, fill = "Spearman's Rho") +
    theme(axis.text.x = element_text(angle = 45, hjust = 1), axis.text.y = element_blank(), axis.ticks.y = element_blank())
  
  return(plot)
}

create_correlation_plots <- function(sig_sum_feat1, sig_sum_feat2) {
  
  # Ensure name and Taxonomy are valid column names
  sig_sum_feat1 <- sig_sum_feat1 %>%
    mutate(name_clean = make.names(name),
           taxonomy_clean = make.names(Taxonomy))
  
  # Add the cleaned columns to sig_sum_feat2
  sig_sum_feat2 <- sig_sum_feat2 %>%
    rename_with(~make.names(.), everything())
  
  purrr::map2(sig_sum_feat1$name_clean, 
              sig_sum_feat1$taxonomy_clean, 
              ~{
                ggplot(data = sig_sum_feat2) +
                  aes(x = .data[[.x]], y = .data[[.y]]) +
                  geom_smooth(method = lm, color = "black", fill = "#A690C4") +
                  geom_text(aes(label = rownames(sig_sum_feat2)), alpha = 0.75) +
                  theme_bw() +
                  labs(x = sig_sum_feat1$name[which(sig_sum_feat1$name_clean == .x)],
                       y = sig_sum_feat1$Taxonomy[which(sig_sum_feat1$taxonomy_clean == .y)]) +
                  annotate("text", -Inf, Inf, 
                           label = bquote("Spearman's \U03C1 ="~.(round(sig_sum_feat1$Rho[which(sig_sum_feat1$taxonomy_clean == .y)], digits = 2))), 
                           hjust = 0, vjust = 1) +
                  #annotate("text", -Inf, Inf, 
                  #         label = bquote("p ="~.(round(sig_sum_feat1$Pval[which(sig_sum_feat1$taxonomy_clean == .y)], digits = 4))), 
                  #         hjust = 0, vjust = 2) +
                  annotate("text", -Inf, Inf, 
                           label = bquote("p (adj) ="~.(round(sig_sum_feat1$padj[which(sig_sum_feat1$taxonomy_clean == .y)], digits = 3))), 
                           hjust = 0, vjust = 2)
              })
}
```

Summarized HMGC279 data first.  
```{r echo=FALSE, message=FALSE, warning=FALSE}
summary_HMGC279_hm = create_heatmap(GUS_species_df, protein_assay_data)
summary_HMGC279_hm
```

The individual features. Y-axis is in reads per million of GUS gene abundances.  
This is showing the features that had a p-value < 0.05 (non-adjusted) and a correaltion coefficient > 0.5.  
```{r echo=FALSE, message=FALSE, warning=FALSE}
sig_sumHMHC279_feat1 = summary_HMGC279_hm$data %>%
  filter(Pval < 0.05 & abs(Rho) > 0.5)

sig_sumHMHC279_feat2 = GUS_species_df %>%
    rownames_to_column(var = "SampleID") %>%
    inner_join(protein_assay_data) %>%
    column_to_rownames("SampleID")

correlation_plots <- create_correlation_plots(sig_sumHMHC279_feat1, sig_sumHMHC279_feat2)
correlation_plots
```

Unsummarized HMGC279 data now.  
```{r echo=FALSE, message=FALSE, warning=FALSE}
HMGC279_hm = create_heatmap(HMGC279, protein_assay_data)
HMGC279_hm
```

The individual features.  
There was one significant feature after BH correction of p-values.  
SRS023176.66822-NC = Mediterraneibacter gnavus.  
```{r echo=FALSE, message=FALSE, warning=FALSE}
sig_HMHC279_feat1 = HMGC279_hm$data %>%
  filter(padj < 0.05 & abs(Rho) > 0.5)

sig_HMHC279_feat2 = HMGC279 %>%
    rownames_to_column(var = "SampleID") %>%
    inner_join(protein_assay_data) %>%
    column_to_rownames("SampleID")

correlation_plots2 <- create_correlation_plots(sig_HMHC279_feat1, sig_HMHC279_feat2)
correlation_plots2
```

HMGI3013 data.  
```{r echo=FALSE, message=FALSE, warning=FALSE}
HMGI3013_hm = create_heatmap(HMGI3013, protein_assay_data)
HMGI3013_hm
```

There were two significant features after BH correction of p-values.  
SRS024549.1421-T1-C = B. fragilis.  
SRS024388.120461-T1-C = B. fragilis also.  
```{r echo=FALSE, message=FALSE, warning=FALSE}
sig_HMGI3013_feat1 = HMGI3013_hm$data %>%
  filter(padj < 0.05 & abs(Rho) > 0.5)

sig_HMGI3013_feat2 = HMGI3013 %>%
    rownames_to_column(var = "SampleID") %>%
    inner_join(protein_assay_data) %>%
    left_join(plot_names) %>%
    column_to_rownames("AltID") %>%
    select(!SampleID)

correlation_plots3 <- create_correlation_plots(sig_HMGI3013_feat1, sig_HMGI3013_feat2)

panel_C = correlation_plots3[[1]] + 
  geom_point(pch = 21, size = 6, color = "black") +
  scale_x_continuous(labels = scales::scientific) +
  scale_y_continuous(labels = scales::scientific) +
  labs(x = "END abundance (Total ion counts)", y = "Reads per million", title = expression(italic("B. fragilis")), subtitle = "SRS024388.120461-T1-C")
panel_C

panel_D = correlation_plots3[[2]] + 
  geom_point(pch = 21, size = 6, color = "black") +
  scale_x_continuous(labels = scales::scientific) +
  scale_y_continuous(labels = scales::scientific) +
  labs(x = "END abundance (Total ion counts)", y = "Reads per million", title = expression(italic("B. fragilis")), subtitle = "SRS024549.1421-T1-C")
panel_D
```

HMGI3013 data.  
Only significant vectors with taxonomy shown. There were many vectors without taxonomy.  
```{r echo=FALSE}
GUS3013_MDS_points = as.data.frame(GUS3013_MDS$points) %>%
  rownames_to_column(var = "SampleID") %>% 
  left_join(plot_names) %>%
  left_join(protein_assay_data)

panel_B = ggplot() +
  geom_segment(data = feature_coords2, 
               aes(x = 0, y = 0, xend = NMDS1, yend = NMDS2),
               arrow = arrow(length = unit(0.2, "cm")), color = "pink", alpha = 0.2) +
  geom_text_repel(data = feature_coords2, 
            aes(x = NMDS1, y = NMDS2, label = `95% Identity Condensation`),
            hjust = -0.2, vjust = 1,  color = "gray50", size = 2.5) +
  geom_point(data = GUS3013_MDS_points, aes(x = MDS1, y = MDS2, fill = `N-Desmethyl-4-hydroxytamoxifen b-D-glucuronide (E/Z mixture)`), pch = 21, color = "black", size = 6) +
  geom_text(data = GUS3013_MDS_points, aes(x = MDS1, y = MDS2, label = AltID)) +
  theme_bw() +
  labs(title = "HMGI3013", x = "MDS1", y = "MDS2", fill = "END \n(Total ion counts)") +
  annotate("text", x = Inf, y = Inf, label = bquote("Stress ="~.(round(GUS3013_MDS$stress, digits = 2))), hjust = 1, vjust = 1) +
  scale_fill_gradient(low = "white", high = "#A690C4", labels = scales::scientific)
```

HMGC279 data.  
Only significant vectors with taxonomy shown.  
```{r echo=FALSE}
GUS279_MDS_points = as.data.frame(GUS279_MDS$points) %>%
  rownames_to_column(var = "SampleID") %>% 
  left_join(plot_names) %>%
  left_join(protein_assay_data) %>%
  filter(complete.cases(.))

panel_E = ggplot() +
  geom_segment(data = feature_coords, 
               aes(x = 0, y = 0, xend = NMDS1, yend = NMDS2),
               arrow = arrow(length = unit(0.2, "cm")), color = "pink", alpha = 0.2) +
  geom_text_repel(data = feature_coords, 
            aes(x = NMDS1, y = NMDS2, label = `cleaned_column`),
            hjust = -0.2, vjust = 1,  color = "gray50", size = 2.5) +
  geom_point(data = GUS279_MDS_points, aes(x = MDS1, y = MDS2, fill = `N-Desmethyl-4-hydroxytamoxifen b-D-glucuronide (E/Z mixture)`), pch = 21, color = "black", size = 6) +
  geom_text(data = GUS279_MDS_points, aes(x = MDS1, y = MDS2, label = AltID)) +
  theme_bw() +
  labs(title = "HMGC279", x = "MDS1", y = "MDS2", fill = "END \n(Total ion counts)") +
  annotate("text", x = Inf, y = Inf, label = bquote("Stress ="~.(round(GUS279_MDS$stress, digits = 2))), hjust = 1, vjust = 1) +
  scale_fill_gradient(low = "white", high = "#A690C4", labels = scales::scientific)
```

Multipanel figure.  
```{r}
genus_GUS_df = GUS_species_df2 %>%
  pivot_longer(!Sample) %>%
  separate(name, into = c("genus", "species", "strain"), remove = F, sep = " ") %>%
  group_by(Sample, genus) %>%
  summarise(value = sum(value))

top_10_gus_genus = genus_GUS_df %>% 
  group_by(genus) %>%
  summarise(mean = mean(value)) %>%
  ungroup %>%
  slice_max(mean, n = 10)
  
GUS_barplot_df = genus_GUS_df %>%
  mutate(genus = if_else(genus %in% top_10_gus_genus$genus, genus, "Other")) %>%
  group_by(Sample, genus) %>%
  summarise(value = sum(value))
  
GUS_barplot_df <- rbind(GUS_barplot_df[!(GUS_barplot_df$genus == "Other"),],GUS_barplot_df[(GUS_barplot_df$genus == "Other"),])
GUS_barplot_df$genus <- factor(GUS_barplot_df$genus, levels = unique(GUS_barplot_df$genus))

GUS_barplot_df = GUS_barplot_df %>%
  filter(!Sample %in% c("2126", "22b", "2621", "Stan")) %>%
  left_join(plot_names, by = join_by("Sample" == "SampleID"))
  
panel_A = ggplot(data = GUS_barplot_df) +
    aes(x = as.character(AltID), weight = value, fill = genus) +
    geom_bar(color = "black") +
    theme_classic() +
    labs(x = NULL, y = "Normalized GUS abundance (RPM)", fill = "Taxonomy", title = "HMGC279") +
    scale_fill_manual(values = Julio_color)
```

```{r}
Figure_7 = ggarrange(
  ggarrange(panel_A, panel_B, ncol = 1, labels = c("A", "B")),
  ggarrange(panel_C, panel_D, nrow = 1, labels = c("C", "D")),
  ncol = 1, heights = c(2,1))

ggsave("Figure_7A.svg", plot = panel_A, width =  6.54, height = 3, dpi = 600, bg = "white", units = "in")
ggsave("Figure_7B.svg", plot = panel_B, width =  6.54, height = 5, dpi = 600, bg = "white", units = "in")
ggsave("Figure_7C.svg", plot = panel_C, width =  3.27, height = 3.27, dpi = 600, bg = "white", units = "in")
ggsave("Figure_7D.svg", plot = panel_D, width =  3.27, height = 3.27, dpi = 600, bg = "white", units = "in")
ggsave("Figure_7E.svg", plot = panel_E, width =  6.54, height = 5, dpi = 600, bg = "white", units = "in")
```
